examples/dmd/wave_equation.cpp

```diff
@@ -0,0 +1,617 @@
+    dmd_training_timer.Start();
+    int curr_window = 0;
+    vector<CAROM::DMD*> dmd_u;
+    dmd_u.push_back(new CAROM::DMD(u.Size(), dt));
+    dmd_u[curr_window]->takeSample(u, t);
+    ts.push_back(t);
+
+    dmd_training_timer.Stop();
+    bool last_step = false;
+    for (int ti = 1; !last_step; ti++)
+    {
+        fom_timer.Start();
+        if (t + dt >= t_final - dt/2)
+        {
+            last_step = true;
+        }
+
+        ode_solver->Step(u, dudt, t, dt);
+        fom_timer.Stop();
+
+        dmd_training_timer.Start();
+        dmd_u[curr_window]->takeSample(u, t);
+
+        if (last_step || (ti % windowNumSamples) == 0)
+        {
+            cout << "step " << ti << ", t = " << t << endl;
+
+            if (rdim != -1) {
+                cout << "Creating DMD with rdim: " << rdim << " at window index: " <<
+                     curr_window << endl;
+                dmd_u[curr_window]->train(rdim);
+            }
+
+            else {
+                cout << "Creating DMD with energy fraction: " << ef << " at window index: " <<
+                     curr_window << endl;
+                dmd_u[curr_window]->train(ef);
+            }
+
+            if (!last_step) {
+                curr_window++;
+                dmd_u.push_back(new CAROM::DMD(u.Size(), dt));
+                dmd_u[curr_window]->takeSample(u, t);
+            }
+
+        }
+        ts.push_back(t);
+        dmd_training_timer.Stop();
+
+        if (last_step || (ti % vis_steps) == 0)
+        {
+
+            u_gf.SetFromTrueDofs(u);
+            dudt_gf.SetFromTrueDofs(dudt);
+
+            if (visualization)
+            {
+                sout << "solution\n" << *mesh << u_gf << flush;
+            }
+
+            if (visit)
+            {
+                visit_dc.SetCycle(ti);
+                visit_dc.SetTime(t);
+                visit_dc.Save();
+            }
+        }
+        oper.SetParameters(u);
+
+    }
+
+    // 9. Save the final solution. This output can be viewed later using GLVis:
+    //    "glvis -m wave_equation.mesh -g wave_equation-final.gf".
+    {
+        ofstream osol("wave_equation-final.gf");
+        osol.precision(precision);
+        u_gf.Save(osol);
+        dudt_gf.Save(osol);
+    }
+
+    // 10. Calculate the DMD modes.
+
+
+    dmd_prediction_timer.Start();
+
+    // 11. Predict the state at t_final using DMD.
+    cout << "Predicting temperature using DMD" << endl;
+
+    CAROM::Vector* result_u = nullptr;
+    VisItDataCollection dmd_visit_dc("DMD_Wave_Equation", mesh);
+    dmd_visit_dc.RegisterField("solution", &u_gf);
+
+    curr_window = 0;
+    if (visit) {
+        result_u = dmd_u[curr_window]->predict(ts[0]);
+        Vector initial_dmd_solution_u(result_u->getData(), result_u->dim());
+        u_gf.SetFromTrueDofs(initial_dmd_solution_u);
+        dmd_visit_dc.SetCycle(0);
+        dmd_visit_dc.SetTime(0.0);
+        dmd_visit_dc.Save();
+    }
+    if (result_u) {
+        delete result_u;
```

Comment:
Perhaps this can be put before line 561 inside `if (visit)`?

Reply:
Done.